pkg/launcher/launcher.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package launcher
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/cilium/cilium/common"
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+var log = common.DefaultLogger
+
+// Launcher is used to wrap the node executable binary.
+type Launcher struct {
+	Mutex   lock.RWMutex
+	target  string
+	arg     string
+	process *os.Process
+	stdout  io.ReadCloser
+}
+
+// Run starts the daemon.
+func (launcher *Launcher) Run() {
+	targetName := launcher.GetTarget()
+	cmd := exec.Command(targetName, launcher.GetArg())
+	cmd.Stderr = os.Stderr
+	stdout, _ := cmd.StdoutPipe()
+	if err := cmd.Start(); err != nil {
+		cmdStr := fmt.Sprintf("%s %s", targetName, launcher.GetArg())
+		log.WithError(err).WithField("cmd", cmdStr).Error("cmd.Start()")
+	}
+
+	launcher.setProcess(cmd.Process)
+	launcher.setStdout(stdout)
+}
+
+// Restart stops the daemon whilauncher will trigger a rerun.
```

Comment:
fix typo "whilauncher"

Reply:
Thanks for looking it over. I tried to split it up into reasonable separate patches for better review, but it is still quite large. I'll fix this up.